reconcile/vault_replication.py

```diff
@@ -0,0 +1,332 @@
+import logging
+import re
+
+from reconcile.utils.vault import (
+    VaultClient,
+    _VaultClient,
+    SecretAccessForbidden,
+    SecretVersionNotFound,
+    SecretNotFound,
+)
+from reconcile.gql_definitions.jenkins_configs import jenkins_configs
+from reconcile.gql_definitions.jenkins_configs.jenkins_configs import (
+    JenkinsConfigV1_JenkinsConfigV1,
+    JenkinsConfigsQueryData,
+)
+
+from reconcile.gql_definitions.vault_policies.vault_policies import (
+    VaultPoliciesQueryData,
+)
+
+from reconcile.gql_definitions.vault_policies import vault_policies
+from reconcile.gql_definitions.vault_instances import vault_instances
+from reconcile.gql_definitions.vault_instances.vault_instances import (
+    VaultInstanceAuthApproleV1,
+    VaultInstanceV1,
+    VaultReplicationConfigV1_VaultInstanceV1_VaultInstanceAuthV1_VaultInstanceAuthApproleV1,
+    VaultReplicationConfigV1_VaultInstanceV1,
+)
+
+from reconcile.utils import gql
+from typing import cast, Optional, Union, Iterable, Mapping
```

Comment:
* `Mapping` and `Iterable` should be imported from `collections.abc`

Reply:
Changed!